### PR TITLE
A small docs update to prepare for v0.3.1-post1

### PR DESCRIPTION
**Pull request by josh146**
This PR updates the documentation, in preparation for changing ReadTheDocs so that it no longer pins to `latest`.

In particular, this PR:

- Cherry picks in the AutoGraph guide 1ab07aac6fd76afca9651138a606bff4d1fcf6ff
- Updates the `@qjit` docstring to reflect new devices
- Adds mention of no `lambda` support to the AutoGraph guide

### CODE REVIEW

File: frontend/catalyst/compilation_pipelines.py [hunk 1]
```diff
@@ -812,16 +812,18 @@ def qjit(
 
     .. note::
 
-        Currently, ``lightning.qubit`` is the only supported backend device
-        for Catalyst compilation. For a list of supported operations, observables,
+        The supported backend devices are ``lightning.qubit``, ``lightning.kokkos``,
+        ``braket.local.qubit``, and ``braket.aws.qubit``.
+        For a list of supported operations, observables,
```
Comment (josh146) [inline frontend/catalyst/compilation_pipelines.py:817]: @maliasadi @dime10 this is the only change in this file not previously reviewed.

File: doc/dev/autograph.rst [hunk 2]
```diff
@@ -0,0 +1,653 @@
...
+Currently, AutoGraph supports converting the following Python control
+flow statements:
+
+- ``if`` statements (including ``elif`` and ``else``)
+- ``for`` loops
+
+``while`` loops are currently not supported, alongside ``break`` and ``continue`` statements.
+
+In addition, ``lambda`` functions cannot currently be used with AutoGraph; please
+use standard function definitions instead.
```
Comment (josh146) [inline doc/dev/autograph.rst:118]: @dime10 @maliasadi this is the only change in this file not previously reviewed

**Comment by josh146** [comment]
> Should we add the while loop support here? I think it's ready to be merged

@dime10 that can be merged into `main`! Once this PR is made and I change the RTDs default pinning.

File: doc/dev/autograph.rst (hunk 2 above)
Comment (dime10) [reply doc/dev/autograph.rst:118]: Looks good, thanks 👍

**Review by dime10 — APPROVED** [review]
Thanks @josh146 👌

**Pull request merged**